infra/base-images/base-runner/test_all.py

```diff
@@ -0,0 +1,176 @@
+# limitations under the License.
+#
+################################################################################
+"""Does bad_build_check on all fuzz targets in $OUT."""
+
+import multiprocessing
+import os
+import re
+import shutil
+import subprocess
+import stat
+import sys
+
+TMP_FUZZER_DIR = '/tmp/not-out'
+
+EXECUTABLE = stat.S_IEXEC | stat.S_IXGRP | stat.S_IXOTH
+
+
+def recreate_directory(directory):
+  """Creates |directory|. If it already exists than deletes it first before
+  creating."""
+  if os.path.exists(directory):
+    shutil.rmtree(directory)
+  os.mkdir(directory)
+
+
+def move_directory_contents(src_directory, dst_directory):
+  """Moves contents of |src_directory| to |dst_directory|."""
+  src_files = os.listdir(src_directory)
+  for filename in src_files:
+    src_path = os.path.join(src_directory, filename)
+    shutil.move(src_path, dst_directory)
+
+
+def find_fuzz_targets(directory):
+  """Returns paths to fuzz targets in |directory|."""
+  fuzzing_language = os.getenv('FUZZING_LANGUAGE')
+  fuzz_targets = []
+  for filename in os.listdir(directory):
+    path = os.path.join(directory, filename)
+    if filename.startswith('afl-'):
+      continue
+    if not os.path.isfile(path):
+      continue
+    if not os.stat(path).st_mode & EXECUTABLE:
+      continue
+    with open(path, 'rb') as file_handle:
+      binary = file_handle.read()
+      if b'LLVMFuzzerTestOneInput' not in binary:
+        continue
+      if fuzzing_language != 'python' and b'ELF' not in binary:
+        continue
+    fuzz_targets.append(path)
+  return fuzz_targets
+
+
+def do_bad_build_check(fuzz_target):
+  """Runs bad_build_check on |fuzz_target|. Returns a
+  Subprocess.ProcessResult."""
+  print('INFO: performing bad build checks for', fuzz_target)
+  command = ['bad_build_check', fuzz_target]
+  return subprocess.run(command,
+                        stderr=subprocess.PIPE,
+                        stdout=subprocess.PIPE,
+                        check=False)
+
+
+def get_broken_fuzz_targets(bad_build_results, fuzz_targets):
+  """Returns a list of broken fuzz targets and their process results in
+  |fuzz_targets| where each item in |bad_build_results| is the result of
+  bad_build_check on the corresponding element in |fuzz_targets|."""
+  broken = []
+  for result, fuzz_target in zip(bad_build_results, fuzz_targets):
+    if result.returncode != 0:
+      broken.append((fuzz_target, result))
+  return broken
+
+
+def has_ignored_targets(out_dir):
+  """Returns True if |out_dir| has any fuzz targets we are supposed to ignore
+  bad build checks of."""
+  out_files = set(os.listdir(out_dir))
+  ignored_targets = {
+      'do_stuff_fuzzer'
+      'checksum_fuzzer', 'fuzz_dump', 'fuzz_keyring', 'xmltest',
+      'fuzz_compression_sas_rle'
+  }
+  if out_files.intersection(ignored_targets):
+    return True
+  for filename in out_files:
+    if re.match(r'ares_*_fuzzer', filename):
```

Comment:
Should ignore list be just a set of regexes ? so can combine with list above ?

Reply:
+1

Reply:
Done. Though it is somewhat complex to do this.